### ISSUE TITLE
[Question] Marquez with authentication 

### ISSUE THREAD

**Issue by TejasRGitHub** [MarquezProject/marquez#2947]
Hi Team, 

I am new to Marquez, and even though Marquez can be deployed AWS - https://marquezproject.ai/docs/deployment/running-on-aws, 

1. Is there any integrated way in which Marquez UI can be put behind any OIDC based authentication ? 
2. Can Marquez be deployed with a firewall or are there configuration which can help doing this ? 
3. Is Marquez designed in such a way that it can be used for any enterprise production system or is it just made as a reference to using Open Lineage ?

**Comment by wslulciuc** [comment]
Thanks for raising this question, @TejasRGitHub! It's a very common question we get a lot, and we recognize that we need to provide a clearer answer or better guidance. I'll will go through your questions one-by-one with as much detail as I can:

> 1. Is there any integrated way in which Marquez UI can be put behind any OIDC based authentication?

Not yet! It's been discuss in https://github.com/MarquezProject/marquez/issues/1137. We've decided to move forward with [shiro](https://shiro.apache.org/index.html) (which does support `OIDC` `SSO`, for example see [_Add Single Sign-on to Your Dropwizard Server in 15 Minutes_](https://developer.okta.com/blog/2018/03/21/dropwizard-oauth)). Auth will be coming in Marquez `0.52.0`.

> 2. Can Marquez be deployed with a firewall or are there configuration which can help doing this?

There's no configuration, but we'd be happy to add any configuration you feel is needed (within reason) to unblock you as we work on introducing shiro in Marquez `0.52.0`. Note, we've seen Marquez UI behind a VPN for internal access, and/or [Auth0](https://auth0.com/docs).

> 3. Is Marquez designed in such a way that it can be used for any enterprise production system or is it just made as a reference to using Open Lineage?

Yes, Marquez does serves as the reference implementation of OpenLineage, but has been deployed as a lineage solution for organizations such as ILUM, Northwestern Mutual, Matillion, Decathlon, and others.

**Comment by TejasRGitHub** [comment]
Hi @wslulciuc , Thanks a lot for answering my question. 

Can you tell me when is Marquez 0.52.0 planned for release ? Can you point me to your roadmap

**Comment by jonathanpmoraes** [comment]
Isn't Shiro a JVM based framework? How is this supposed to work with React for the Marquez UI?

**Comment by TejasRGitHub** [comment]
Hi @wslulciuc , could you please tell if the release ( https://github.com/MarquezProject/marquez/milestone/18 ) would still be coming on 2024-12-18 or has it been postponed to a different timeline.

**Comment by tib-rijo** [comment]
> Note, we've seen Marquez UI behind a VPN for internal access, and/or [Auth0](https://auth0.com/docs).

For people coming here with an interest to find interim workarounds that do not depend on availability of a (non-commercial) VPN, such as WireGuard or Headscale/Tailscale, or a (non-commercial) identity and access proxy, such as [pomerium](https://github.com/pomerium/pomerium) or [authelia](https://github.com/authelia/authelia), or a mix between the two, such as [octelium](https://github.com/octelium/octelium) or [teleport](https://github.com/gravitational/teleport), you can always try out to put

- [oauth2-proxy/oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy)

in front of your application. It also supports [Træfik Forward Auth](https://github.com/oauth2-proxy/oauth2-proxy).
